monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+        # calculated separately
+        self.SUMMERY_AVERAGE = average
+        self.DO_CONNECTED_COMP = do_ccp
+        self.device = device
+        # gather all summary function for combining case stats
+        self.gather_summary: Dict[str, Dict] = {}
+        self._hardcode_functions()
+        self._hardcode_operations()
+        print(self.gather_summary)
+
+    def _hardcode_functions(self):
+        """Register functions for calculating stats."""
+        self.functions = [self._get_case_image_stats, self._get_case_foreground_image_stats, self._get_label_stats]
+        self.functions_summary = [
+            self._get_case_image_stats_summary,
+            self._get_case_foreground_image_stats_summary,
+            self._get_label_stats_summary,
+        ]
+
+    def _hardcode_operations(self):
+        """Register operations for basic data operation."""
+        # define basic operations for stats
+        self.operations = {
+            "max": torch.max,
+            "mean": torch.mean,
+            "median": torch.median,
+            "min": torch.min,
+            "percentile": partial(torch.quantile, q=[0.005, 0.10, 0.90, 0.995]),
+            # 'percentile_00_5' : partial(torch.quantile, q=.005),
+            # 'percentile_99_5' : partial(torch.quantile, q=.995),
+            # 'percentile_10_0' : partial(torch.quantile, q=.10),
+            # 'percentile_90_0' : partial(torch.quantile, q=.90),
+            "stdev": partial(torch.std, unbiased=False),
+        }
+        # allow mapping the output of self.operations to new keys (save computation time)
+        # the output from torch.quantile is mapped to four keys.
+        self.operations_mappingkey = {
+            "percentile": ["percentile_00_5", "percentile_10_0", "percentile_90_0", "percentile_99_5"]
+        }
+        # np version of self.operations. torch operation has inconsistent interfaces.
+        # only used in the summary part.
+        self.operations_np = {
+            "max": np.max,
+            "mean": np.mean,
+            "median": np.median,
+            "min": np.min,
+            "percentile": partial(np.quantile, q=[0.005, 0.10, 0.90, 0.995]),
+            # 'percentile_00_5' : partial(np.percentile, q=0.5),
+            # 'percentile_99_5' : partial(np.percentile, q=99.5),
+            # 'percentile_10_0' : partial(np.percentile, q=10),
+            # 'percentile_90_0' : partial(np.percentile, q=90),
+            "stdev": np.std,
+        }
+        # define summary functions for output from self.operations. For exmaple,
+        # how to combine max intensity from each case (image)
+        self.operations_summary = {
+            "max": np.max,
+            "mean": np.mean,
+            "median": np.mean,
+            "min": np.min,
+            "percentile_00_5": np.mean,
+            "percentile_99_5": np.mean,
+            "percentile_10_0": np.mean,
+            "percentile_90_0": np.mean,
+            "stdev": np.mean,
```

Comment:
Same comment to define Enum for the keys.

Thanks.

Reply:
Thanks @Nic-Ma,  I am thinking if whether we should either define the operations in Enum or temporarily put this into backlog for now. For code readability, what may be even better is to define the keys in the dataset summary output in a Enum rather than the operation. Then the developer would know the structure of the output. But it will be quite some work on the refracting, and I would suggest to put it in backlog for now.

Reply:
OK, sounds good to me.

Thanks.